### PR TITLE
handle case of missing image.config.openshift.io type definition

### PR DESCRIPTION
**Pull request by deads2k**
When the entire CRD is missing, we can still synchronize most of the config and simply use the last known value for the image config itself.  We need to observe most of the config before this loop starts.

/assign @sanchezl @juanvallejo @sttts 

@sanchezl can you manually verify that if no CRD is present when this process starts, adding the CRD later and setting a value is properly observed?  If not, please add a unit test showing it breaking and find a way to fix it.

@bparees note the structural change

### CODE REVIEW

File: pkg/operator/observe_config.go [hunk 1]
```diff
@@ -280,28 +280,34 @@ func observeRestrictedCIDRs(listers Listers, currentConfig map[string]interface{
 
 // observeInternalRegistryHostname reads the internal registry hostname from the cluster configuration as provided by
 // the registry operator.
-func observeInternalRegistryHostname(listers Listers, currentConfig map[string]interface{}) (observedConfig map[string]interface{}, errs []error) {
-	observedConfig = map[string]interface{}{}
+func observeInternalRegistryHostname(listers Listers, existingConfig map[string]interface{}) (map[string]interface{}, []error) {
```
Comment (deads2k) [inline pkg/operator/observe_config.go]: @sanchezl we don't use named return values.  Please update the others.

File: pkg/operator/observe_config.go [hunk 2]
```diff
@@ -280,28 +280,34 @@ func observeRestrictedCIDRs(listers Listers, currentConfig map[string]interface{
...
+	errs := []error{}
+	prevObservedConfig := map[string]interface{}{}
+
 	internalRegistryHostnamePath := []string{"imagePolicyConfig", "internalRegistryHostname"}
-	currentInternalRegistryHostname, _, _ := unstructured.NestedStringSlice(currentConfig, internalRegistryHostnamePath...)
+	if currentInternalRegistryHostname, _, _ := unstructured.NestedString(existingConfig, internalRegistryHostnamePath...); len(currentInternalRegistryHostname) > 0 {
+		unstructured.SetNestedField(prevObservedConfig, currentInternalRegistryHostname, internalRegistryHostnamePath...)
```
Comment (deads2k) [inline pkg/operator/observe_config.go]: @sanchezl this structure makes it a little easier to follow the logic of "return the previous value"

File: pkg/operator/observe_config.go [hunk 3]
```diff
@@ -89,14 +88,15 @@ func NewConfigObserver(
 			imageConfigLister: configInformer.Config().V1().Images().Lister(),
 			endpointsLister:   kubeInformersForKubeSystemNamespace.Core().V1().Endpoints().Lister(),
 			configmapLister:   kubeInformersForKubeSystemNamespace.Core().V1().ConfigMaps().Lister(),
+			imageConfigSynced: configInformer.Config().V1().Images().Informer().HasSynced,
+		},
+		cachesSynced: []cache.InformerSynced{
```
Comment (deads2k) [inline pkg/operator/observe_config.go]: notice that the optional types aren't listed here.

**Comment by bparees** [comment]
lgtm.

File: pkg/operator/observe_config.go (hunk 2 above)
Comment (sanchezl) [reply pkg/operator/observe_config.go]: :+1:

**Pull request merged**